src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -321,6 +321,14 @@ private static IEnumerable<string> ReadLanguagesFromAttribute(ref BlobReader arg
             return SpecializedCollections.EmptyEnumerable<string>();
         }
 
+        private static ISourceGenerator? CoerceGeneratorType(object? generator)
+        {
+            if (generator is IIncrementalGenerator)
+            {
+                return (ISourceGenerator?)Activator.CreateInstance(typeof(IncrementalToSourceGeneratorWrapper<>).MakeGenericType(generator.GetType()), generator);
```

Comment:
```suggestion
            if (generator is IIncrementalGenerator ig)
            {
                return new IncrementalToSourceGeneratorWrapper<IIncrementalGenerator>(ig);
```

Why is this not sufficient here? Basically what is the value in having the generic parameter be the actual generator type vs. just using the interface here given we can't take advantage of the concrete type in the compiler?